### ISSUE TITLE
Just run one check at the time!

### ISSUE THREAD

**Issue by Nubro01** [Tribler/tribler#3172]
- [ ] I have tried with the latest pre-release version and I still can reproduce the issue.
Oct 20th builtd
##### Tribler version/branch+revision: #####
Tribler 7
##### Operating system and version: #####
Windows 10 Insider 17017
##### Steps to reproduce the behavior: #####
Start multiple Torrent checks
##### Expected behavior: #####
If same location then que the checking.
##### Actual behavior: #####
all checks are started at the same time
##### Relevant log file output: #####
When 3 checks are running, where ALL files that are checked are on the same drive, then within Windows taskmanager the drive is at 100% and disk read is just 6MBs.
When I stop two, the drive goes to 54% and the read speed is 100 MBs, so much faster, which is logical, because the files are spread over the disk, so increased head movement, slows down the whole process.
I do not know if you guys have access to the destination/location, but if you have to check a Torrent and one is already running, check if this new check is in the same location, if it is, que it. if it is not, then start it.

**Comment by qstokkink** [comment]
We should probably make this configurable. Something like a `Maximum parallel checks` setting.

**Comment by ichorid** [comment]
I would say Tribler should only check a single torrent at a time by default, though the setting to disable this limitation should be available in the core config file. Checking lots of torrents in parallel makes no sense for all but fastest SSDs because the checking procedure is always i/o bound.

**Comment by devos50** [comment]
Not sure whether parallel torrent checking is IO bound. I think it is network bound since getting a response from the DHT usually take up to a few second. Since we only use a single UDP socket for lookups, we also do not keep multiple sockets open. We should aim for some performance experiments with torrent lookups.

**Comment by ichorid** [comment]
@devos50 , by "torrent checking" in this thread we mean "checking the files on the disk against the data in the torrent file", i.e. rehashing. It has nothing to do with the network at all :wink:

**Comment by Nubro01** [comment]
The latest test version nr 7.13 shows stopped Torrents as white and 0% in the progress bar. When you start them they start checking, while that is not needed! If you start them all at ones, they all start checking, while active torrents are also seeding from the same disk. 3 tasks is already slowing things down on a hard disk and although an SSD might have 4 channels a harddisk has not, so things really slow down. Please create a que for torrents to be checked and do not start them all at once!

**Comment by Nubro01** [comment]
Last night I was downloading en windows 11 23h2 update, made the computer reboot. This morning at restart of the latest Tribler version, 30 of my 150 library instances where in checking status. As they all are on the same disk, everything was dead slow.
Stopped them all manually and restarted one by one! Please fix this, it is techinally wrong to check all files that are in the same location!

**Comment by absolutep** [comment]
I think this is not a very pressing issue at the moment and can be transferred to 8.3

**Comment by qstokkink** [comment]
@absolutep You're right. However, we've been trying to get this done for 8 years. At this point, this one is personal. Against all rationality, we just really _want_ to get this done.

**Comment by teddyrogers** [comment]
It is an annoyance! You want hash checking but don't want the inconvenience of this bug occurring and bogging down your system for a few hours.

My suggestion would be once the download has completed and hash check has run, and all files are okay, add a flag in the checkpoint file. The only time a hash check needs to be performed again would be if the user manually prompts this.

**Comment by absolutep** [comment]
> [@absolutep](https://github.com/absolutep) You're right. However, we've been trying to get this done for 8 years. At this point, this one is personal. Against all rationality, we just really _want_ to get this done.

I perfectly understand this sentiment.

I just wanted to get my hands on latest stable 8.2 ASAP.  :)
